src/aws-cpp-sdk-core/source/Version.cpp

```diff
@@ -36,15 +36,44 @@ namespace Version
 #define xstr(s) str(s)
 #define str(s) #s
 #if defined(_MSC_VER)
-      return "MSVC/" xstr(_MSC_VER);
+      return "MSVC#" xstr(_MSC_VER);
 #elif defined(__clang__)
-      return "Clang/" xstr(__clang_major__) "."  xstr(__clang_minor__) "." xstr(__clang_patchlevel__);
+      return "Clang#" xstr(__clang_major__) "."  xstr(__clang_minor__) "." xstr(__clang_patchlevel__);
 #elif defined(__GNUC__)
-      return "GCC/" xstr(__GNUC__) "."  xstr(__GNUC_MINOR__) "." xstr(__GNUC_PATCHLEVEL__);
+      return "GCC#" xstr(__GNUC__) "."  xstr(__GNUC_MINOR__) "." xstr(__GNUC_PATCHLEVEL__);
 #else
       return "UnknownCompiler";
 #endif
 #undef str
+#undef xstr
+  }
+
+  const char* GetCPPStandard()
+  {
+#define xstr(s) str(s)
+#define str(s) #s
+#if defined(CPP_STANDARD)
```

Comment:
i know `CPP_STANDARD` is used internally to [specify the standard flag](https://github.com/aws/aws-sdk-cpp/blob/05edc68675ba391e8669641ffa4e034ba691cb97/cmake/platform/linux.cmake#L38)

```
 list(APPEND CMAKE_REQUIRED_FLAGS "-std=c++${CPP_STANDARD}")
```

but do we know how this value intersects with the cmake arg [`CMAKE_CXX_STANDARD`](https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html)? This is likely perhaps outside of scope of this PR. But it is something we should likely reconcile at some point.

Reply:
`__cplusplus` shall be a definitive answer, while CPP_STANDARD is a cmake argument that we allow to configure.
You are right we might need to revisit this later.